win32ss/drivers/videoprt/videoprt.c

```diff
@@ -526,6 +524,12 @@ IntVideoPortFindAdapter(
         goto Failure;
     }
 
+    if(DriverExtension->InitializationData.HwInterrupt != NULL)
+    {
+        ConfigInfo.BusInterruptLevel = DeviceExtension->InterruptLevel;
+        ConfigInfo.BusInterruptVector = DeviceExtension->InterruptVector;
```

Comment:
You're still erasing what the miniport driver has set into the ConfigInfo.BusInterrupt*** members with something else. This is wrong!

Reply:
> BusInterruptLevel
> 
> The bus-relative interrupt request level. Storport makes no assumptions about the HBA's interrupt usage, so the default value is zero. Storport initializes this member and miniport drivers must not modify it.

Reply:
@HBelusca As yo can see miniport must not modify this

Reply:
But you're reading me the documentation of the storage ports, **AND NOT** the one of the **video** miniport drivers :rofl:

https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/video/ns-video-_video_port_config_info
clearly says:
```
BusInterruptLevel

This member is irrelevant if the video hardware does not generate interrupts, indicated by setting the HwInterrupt member in the [VIDEO_HW_INITIALIZATION_DATA](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/video/ns-video-_video_hw_initialization_data) structure to NULL. Otherwise, it specifies the bus-relative IRQL that corresponds to the interrupt request on Isa or MicroChannel type buses. _**The preset default value for this member is zero. A miniport driver must supply the correct value if it handles interrupts for a video adapter on an Isa or MicroChannel type bus, or for a video adapter configured for level-sensitive interrupts on an Eisa type bus.**_

If a miniport driver's [HwVidFindAdapter](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/video/nc-video-pvideo_hw_find_adapter) function finds that the video adapter does not generate interrupts or that it cannot determine a valid interrupt vector/level for the adapter, _**HwVidFindAdapter should set both BusInterruptLevel and BusInterruptVector to zero.**_

BusInterruptVector

This member is irrelevant if the video hardware does not generate interrupts, indicated by setting the HwInterrupt member in the VIDEO_HW_INITIALIZATION_DATA structure to NULL. Otherwise, it specifies the bus-relative vector used by the video hardware on I/O buses that use interrupt vectors, such as PCI buses. The preset default value for this member is zero.

If a miniport driver's HwVidFindAdapter function finds that the video adapter does not generate interrupts or that it cannot determine a valid interrupt vector/level for the adapter, HwVidFindAdapter should set both BusInterruptVector and BusInterruptLevel to zero.
```

(Also, searching for legacy video miniports, e.g. those in the NT4 DDK, would show you _actual_ usage examples.)

Reply:
@HBelusca I'm reading windows 2000 samples drivers and headers. The sample drivers that uses interrupts don't touch the BysInterruptVector and Level, maybe the behaviour has changed since then?

Reply:
@HBelusca the latest changes should make the difference